src/box/iproto.cc

```diff
@@ -1520,6 +1520,8 @@ tx_finish_request(struct cmsg* m)
 	struct session *session = msg->connection->session;
 	session->requests_count--;
 	assert(session->requests_count >= 0);
+	if (is_shutdown_ready(session))
```

Comment:
I see that this function `tx_finish_request` calling for every request. Also for simple requests and for not `graceful shutdown` connections `is_shutdown_ready` is very very often true. I'm afraid of losing performance. I think we need to make sure that this does not happen.